packages/batch-nft-reveal/contracts/NFTCollection.sol

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@chainlink/contracts/src/v0.8/VRFConsumerBase.sol";
+import "@chainlink/contracts/src/v0.8/KeeperCompatible.sol";
+import "base64-sol/base64.sol";
+import "./interfaces/IERC20.sol";
+
+contract NFTCollection is
+    Ownable,
+    ERC721,
+    VRFConsumerBase,
+    KeeperCompatibleInterface
+{
+    // STRUCTS
+
+    struct Metadata {
+        uint256 startIndex;
+        uint256 endIndex;
+        uint256 entropy;
+    }
+
+    // IMMUTABLE STORAGE
+
+    uint256 public immutable maxSupply;
+    uint256 public immutable mintCost;
+    address public immutable linkToken;
+    uint256 internal immutable linkFee;
+    bytes32 internal immutable keyHash;
+
+    // MUTABLE STORAGE
+
+    uint256 public totalSupply = 0;
+    uint256 public revealedCount = 0;
+    uint256 public revealBatchSize;
+    uint256 public revealInterval;
+    uint256 public lastRevealed;
+    bool public pendingReveal = false;
+    Metadata[] public metadatas;
+
+    // ERRORS
+
+    error InvalidAmount();
+    error MaxSupplyReached();
+    error InsufficientFunds();
+    error RevealCriteriaNotMet();
+    error InsufficientLINK();
+    error WithdrawProceedsFailed();
+
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        uint256 _maxSupply,
+        uint256 _mintCost,
+        uint256 _revealBatchSize,
+        uint256 _revealInterval,
+        address _vrfCoordinator,
+        address _linkToken,
+        uint256 _linkFee,
+        bytes32 _linkKeyHash
+    ) ERC721(_name, _symbol) VRFConsumerBase(_vrfCoordinator, _linkToken) {
+        maxSupply = _maxSupply;
+        mintCost = _mintCost;
+        revealBatchSize = _revealBatchSize;
+        revealInterval = _revealInterval;
+        linkToken = _linkToken;
+        linkFee = _linkFee;
+        keyHash = _linkKeyHash;
+    }
+
+    // ACTIONS
+
+    function mint(uint256 _amount) external payable {
+        if (_amount <= 0) {
```

Comment:
```suggestion
```


_amount is uint, I don't think we need to check if its less than 0. When i tried to pass mint() negative value it threw automatically an error.